src/app/common/filter-text/filter-text.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class FilterTextService {
+  constructor() {}
+
+  filter(data: string, props: Array<string>, originalList: Array<any>) {
+    let filteredList = [];
+
+    if (data && props && originalList) {
+      data = data.toLowerCase();
+
+      originalList.filter(item => {
+        let match = false;
+
+        for (const prop of props) {
+
+          console.log('prop: ' + JSON.stringify(prop));
```

Comment:
Be good to take out these console.log statements, or we should perhaps use a proper logging subsystem so we don't clutter up the dev console